projects/igniteui-angular/src/lib/query-builder/query-builder-drag.service.ts

```diff
@@ -0,0 +1,532 @@
+                    this.targetElement = this._dropZonesList[index + indexOffset]
+                    this.targetExpressionItem = this._expressionsList[index + indexOffset];
+                } else {
+                    //if the current drop zone is a group root (AND/OR)
+                    if (index + indexOffset === 0) {
+                        //If the root group's AND/OR
+                        this.targetElement = this._dropZonesList[0]
+                        this.targetExpressionItem = this._queryBuilderTreeComponent.rootGroup.children[0];
+                        under = true;
+                    } else if (under) {
+                        //If under AND/OR
+                        this.targetElement = this._dropZonesList[index + indexOffset]
+                        this.targetExpressionItem = this._expressionsList[index + indexOffset + 1];
+                    } else {
+                        //if over AND/OR
+                        this.targetElement = this._dropZonesList[index + indexOffset].parentElement.parentElement;
+                        this.targetExpressionItem = this._expressionsList[index + indexOffset];
+                    }
+
+                    //If should drop under AND/OR => drop over first chip in that AND/OR's group
+                    if (under) {
+                        this.targetElement = this.targetElement.nextElementSibling.firstElementChild as HTMLElement;
+                        if (this.targetElement === this.dropGhostChipNode) this.targetElement = this.targetElement.nextElementSibling as HTMLElement;
+                        under = false;
+                    }
+                }
+                const before = this.getPreviousChip(this.dropGhostElement);
+                const after = this.getNextChip(this.dropGhostElement);
+
+                this.renderDropGhostChip(this.targetElement, under, true);
+
+                //If it's the first arrow hit OR drop ghost is not displayed OR hasn't actually moved, move one more step in the same direction
+                if (this._keyDragFirstMove ||
+                    !this.dropGhostElement ||
+                    (this.getPreviousChip(this.dropGhostElement) === before && this.getNextChip(this.dropGhostElement) === after)) {
+                    this._keyDragFirstMove = false;
+                    this.arrowDrag(key);
+                }
+            } else {
+                //Dropping on '+ Condition button' => drop as last condition in the root group
+                let lastElement = this._dropZonesList[this._dropZonesList.length - 1].parentElement.previousElementSibling
+                if (lastElement.className.indexOf("igx-filter-tree__expression-section") !== -1) lastElement = lastElement.lastElementChild;
+                if (lastElement.className.indexOf("igx-filter-tree__subquery") !== -1) lastElement = lastElement.previousElementSibling
```

Comment:
These strings could be added to the QueryBuilderConstants and used from there. This way if they are changed at some point it would be easier to fix